scripts/s3manage.py

```diff
@@ -0,0 +1,568 @@
+
+
+def version_info(version):
+    info = get_version_info(version)
+    if info is None:
+        print "No info for version {}".format(version)
+        sys.exit(1)
+    for k in info.keys():
+        print "{}: {}".format(k, info[k])
+
+
+def delete_version(version):
+    if version_exists(version) is False:
+        print("Version '{}' does not exists in AWS S3. Aborting".format(version))
+        sys.exit(2)
+
+    if version == get_active_version():
+        print("Version '{}' is the active version. You cannot delete it".format(version))
+        sys.exit()
+
+    files = bucket.objects.filter(Prefix=str(version)).all()
+
+    indexes = [{'Key': k.key} for k in files]
+    for n in ('index', 'embed', 'mobile'):
+        src_key_name = '{}.{}.html'.format(n, version)
+        indexes.append({'Key': src_key_name})
+
+    resp = s3client.delete_objects(Bucket=BUCKET_NAME, Delete={'Objects': indexes})
+
+    for v in resp['Deleted']:
+        print v
+
+
+def activate(version):
+    if version_exists(version) is False:
+        print("Version '{}' does not exists in AWS S3. Aborting".format(version))
+        sys.exit(2)
+
+    if version == get_active_version():
+        print("Version '{}' is already the active version. Doing nothing".format(version))
+        sys.exit()
+
+    # Source files
+
+    files = bucket.objects.filter(Prefix='{}/src'.format(version)).all()
+
+    for k in files:
+        src_key_name = k.key
+        dst_key_name = src_key_name.replace('{}/'.format(version), '')
+        print("{} --> {}".format(src_key_name, dst_key_name))
+
+        s3client.copy_object(
+            Bucket=BUCKET_NAME,
+            CopySource=BUCKET_NAME +
+            '/' +
+            src_key_name,
+            Key=dst_key_name,
+            ACL='public-read')
+
+    # Prod files
+    for n in ('index', 'embed', 'mobile'):
+        src_key_name = '{}.{}.html'.format(n, version)
+        print("{} --> {}.html".format(src_key_name, n))
+
+        s3client.copy_object(
+            Bucket=BUCKET_NAME,
+            CopySource=BUCKET_NAME +
+            '/' +
+            src_key_name,
+            Key=n +
+            '.html',
+            ACL='public-read')
+    print "Special files"
+    appcache = None
+    files = list(bucket.objects.filter(Prefix='{}/geoadmin.'.format(version)).all())
+    if len(files) > 0:
+        appcache = os.path.basename(sorted(files)[-1].key)
+    for j in ('robots.txt', 'checker', appcache):
+        src_key_name = '{}/{}'.format(version, j)
+        print src_key_name, os.path.basename(src_key_name)
+        try:
+            s3client.copy_object(
+                Bucket=BUCKET_NAME,
+                CopySource=BUCKET_NAME + '/' + src_key_name,
+                Key=os.path.basename(src_key_name),
+                CopySourceIfModifiedSince=datetime(2015, 1, 1),
+                ACL='public-read')
+        except botocore.exceptions.ClientError as e:
+            print "Cannot copy {}: {}".format(j, e)
+
```

Comment:
Can you explain more in details what are those `Special files`.  The only difference here as far as I can tell is `CopySourceIfModifiedSince=datetime(2015, 1, 1)` which is an hardcoded datetime.